src/IB/IBFESurfaceMethod.cpp

```diff
@@ -1229,10 +1952,78 @@ IBFESurfaceMethod::initializeFEEquationSystems()
                 F_system.add_variable("F_" + std::to_string(d), d_fe_order[part], d_fe_family[part]);
             }
 
-            if (d_use_jump_conditions)
+            if (d_use_pressure_jump_conditions)
             {
-                auto& DP_system = equation_systems->add_system<System>(PRESSURE_JUMP_SYSTEM_NAME);
-                DP_system.add_variable("C_p", d_fe_order[part], d_fe_family[part]);
+                System& P_jump_system = equation_systems->add_system<System>(PRESSURE_JUMP_SYSTEM_NAME);
+                System& P_system = equation_systems->add_system<System>(P_SYSTEM_NAME);
+                if (d_use_l2_lagrange_family)
+                {
+                    P_jump_system.add_variable("P_jump_", d_fe_order[part], L2_LAGRANGE);
+                    P_system.add_variable("P_", d_fe_order[part], L2_LAGRANGE);
+                }
+                else
+                {
+                    P_jump_system.add_variable("P_jump_", d_fe_order[part], d_fe_family[part]);
+                    P_system.add_variable("P_", d_fe_order[part], d_fe_family[part]);
+                }
+            }
+
+            if (d_use_velocity_jump_conditions)
+            {
+                std::array<System*, NDIM> DU_jump_system;
+                for (unsigned int d = 0; d < NDIM; ++d)
+                {
+                    DU_jump_system[d] = &equation_systems->add_system<System>(VELOCITY_JUMP_SYSTEM_NAME[d]);
+                    for (unsigned int i = 0; i < NDIM; ++i)
+                    {
+                        const std::string system_name = "DU_jump_" + std::to_string(d) + "_" + std::to_string(i);
+                        // TODO: The value of the FEFamily should just be read in from the input file directly instead
+                        // of using a boolean.
+                        if (d_use_l2_lagrange_family)
+                        {
+                            DU_jump_system[d]->add_variable(system_name, d_fe_order[part], L2_LAGRANGE);
+                        }
+                        else
+                        {
+                            DU_jump_system[d]->add_variable(system_name, d_fe_order[part], d_fe_family[part]);
+                        }
```

Comment:
This logic seems overly complicated both here and below. Can we just resolve this TODO now?